src/components/Table/Table.Body.tsx

```diff
@@ -10,8 +10,11 @@ export default class Body extends React.Component<BodyProps, BodyState> {
   columnsCache = this.getColumnsCache(this.props.columns)
 
   static getDerivedStateFromProps(nextProps, prevState) {
+    /* istanbul ignore else */
     if (!equal(nextProps.rows, prevState.rows)) {
       return { rows: Body.getRows(nextProps) }
+    } else if (nextProps.isTableCollapsed) {
+      return { rows: Body.getRows(nextProps) }
     }
```

Comment:
since they both do the same actions, should we just have one `if` statement?

```suggestion
    const rowsAreNotEquals = !equal(nextProps.rows, prevState.rows) 
    if (rowsAreNotEquals || nextProps.isTableCollapsed) {
      return { rows: Body.getRows(nextProps) }
    }
```

Reply:
Ah yeah nice catch :)